apps/voting/app/src/vote-hooks.js

```diff
@@ -0,0 +1,80 @@
+import { useState, useEffect } from 'react'
+import { useAragonApi } from '@aragon/api-react'
+
+// when `params` are updated, call `fn` and pass the result
+function usePromiseResult(fn, params, memoParams, defaultValue) {
+  const [result, setResult] = useState(defaultValue)
+  useEffect(() => {
+    let cancelled = false
+    fn(...params).then(value => {
+      if (!cancelled) {
+        setResult(value)
+      }
+    })
+    return () => {
+      cancelled = true
+    }
+  }, memoParams)
+  return result
+}
+
+async function getUserBalance(vote, userAccount, tokenContract, tokenDecimals) {
+  if (!tokenContract || !userAccount) {
+    return -1
```

Comment:
It seems like the checks in `VotePanelContent` usually check if this is `null` instead of `-1`. We should make sure to change those checks to use `-1`.

Should we also add a check for `!vote`?